### ISSUE TITLE
upgrade to ivy 2.3.0 (final)

### ISSUE THREAD

**Issue by jdolson** [sbt/sbt#847]
sbt 0.13.0 is still using ivy 2.3.0-rc1 (released Apr. 2012). The final was [released](http://ant.apache.org/ivy/history/2.3.0/release-notes.html) in Jan. 2013 and includes several fixes not in rc1 (or even rc2). The second RC broke two public APIs sbt uses and so the fix is nontrivial. It is too late to fix this for 0.13 but it could be fixed for 0.14.

**Comment by zavg** [comment]
The sbt 0.13.0 is still depends on ivy 2.3.0-rc1, which, for example, consists critical bug [[IVY-1378]](https://issues.apache.org/jira/browse/IVY-1378) which does not allow run sbt on Windows if the name of current user contains some non-ascii characters (Umlauts, cyrillic symbols, etc.)

**Comment by peter-svensson** [comment]
IVY 2.3.0-rc1 also contains a bug that handles transitive dependency versions wrong. 

Dependency:

```
libraryDependencies ++= Seq(
  "org.glassfish.jersey.core" % "jersey-client" % "2.4.1"
)
```

When resolving the transitive dependencies of that jar I get errors:

```
org.glassfish.hk2#hk2-utils;2.4.1: not found
org.glassfish.hk2#hk2-locator;2.4.1: not found
```

Looking at the dependency tree I can see that I get the following:

```
[info]   +-org.glassfish.jersey.core:jersey-client:2.4.1
[info]     +-com.google.guava:guava:14.0.1
[info]     +-javax.ws.rs:javax.ws.rs-api:2.0
[info]     +-org.glassfish.hk2.external:javax.inject:2.2.0-b21
[info]     +-org.glassfish.hk2:hk2-api:2.2.0-b21
[info]     | +-org.glassfish.hk2:hk2-utils:2.4.1 (error: not found)
[info]     | 
[info]     +-org.glassfish.hk2:hk2-locator:2.4.1 (error: not found) 
```

Looking in the pom.file for hk2-api (which is the file that requires hk2-utils) [http://repo1.maven.org/maven2/org/glassfish/hk2/hk2-api/2.2.0-b21/hk2-api-2.2.0-b21.pom] we see that:

```
<dependencies>
...
<dependency>
<groupId>org.glassfish.hk2</groupId>
<artifactId>hk2-utils</artifactId>
<version>${project.version}</version>
</dependency>
</dependencies>
```

And the project.version refers to the version of hk2-api which is defined in [http://repo1.maven.org/maven2/org/glassfish/hk2/hk2-public/2.2.0-b21/hk2-public-2.2.0-b21.pom]

So the correct version should be 2.2.0-b21 for hk2-utils and not 2.4.1 which is clearly from the originating dependency (jersey-client).

I verified that the bug is fixed in ivy 2.3.0 (and it also worked in 2.2.0....) so an upgrade of SBT to use IVY 2.3.0 would be very good :)

**Comment by harrah** [comment]
Pull request #1040 pending.  Please comment there if you have an opinion on the risk/value of merging in 0.13.2 v. 0.14.0.

**Comment by harrah** [comment]
Fixed by #1040.
